app/controllers/green_lanes/applicable_exemptions_controller.rb

```diff
@@ -7,24 +7,58 @@ class ApplicableExemptionsController < ApplicationController
                   :disable_search_form
 
     def new
-      @exemptions_form = exemptions_form
+      @exemptions_form = build_exemptions_form
+      @back_link_path = determine_back_link_path
 
       render_exemptions_questions
     end
 
     def create
-      @exemptions_form = exemptions_form
+      @exemptions_form = build_exemptions_form
 
       if @exemptions_form.valid?
         next_page = determine_next_page
         redirect_to handle_next_page(next_page)
       else
+        @back_link_path = determine_back_link_path
+
         render_exemptions_questions
       end
     end
 
     private
 
+    def determine_back_link_path
+      permitted_params = params.permit(:commodity_code, :country_of_origin, :moving_date, :c1ex, :c2ex, ans: {})
+
+      if category == 2
+        new_green_lanes_applicable_exemptions_path(
+          category: 1,
+          commodity_code: permitted_params[:commodity_code],
+          country_of_origin: permitted_params[:country_of_origin],
+          moving_date: permitted_params[:moving_date],
+          ans: parsed_ans(permitted_params[:ans]),
+          c1ex: permitted_params[:c1ex],
+        )
+      else
+        new_green_lanes_moving_requirements_path(
+          commodity_code: params[:commodity_code],
+          country_of_origin: params[:country_of_origin],
+          moving_date: params[:moving_date],
+        )
+      end
+    end
+
+    def parsed_ans(ans_param)
+      return ans_param unless ans_param.is_a?(String)
+
+      begin
+        JSON.parse(ans_param)
+      rescue JSON::ParserError
+        ans_param
```

Comment:
so if you get an error you return the `parsed_ans` unparsed?
why is that?

Reply:
Yeah possibly i can skip this part because its virtually impossible to happen will remove